hadoop-hdfs-project/hadoop-hdfs/src/main/java/io/hops/transaction/lock/LeaseLock.java

```diff
@@ -89,4 +99,16 @@ protected void acquire(TransactionLocks locks) throws IOException {
   protected final Type getType() {
     return Type.Lease;
   }
+
+  public void setSingleFileLock(String path){
```

Comment:
Wouldn't it be better to set it in the constructor rather than have a setter that set it afterward?

Reply:
Actually in both cases the code looks horrible. I should be a new lock class.

Reply:
I take back the idea of new class. New constructor it is

Reply:
new constructor it si